cluster-autoscaler/cloudprovider/packet/packet_manager_rest.go

```diff
@@ -27,17 +27,140 @@ import (
 	"math/rand"
 	"net/http"
 	"os"
+	"strings"
```

Comment:
This whole file seems quite duplicative. There is an [official packet go sdk](https://godoc.org/github.com/packethost/packngo). All of the work being done here duplicates it and requires maintenance. Why is this large file here instead of just leveraging packngo?

Reply:
As for the packngo package I would be glad to use it, but as it has been mentioned in the first PR of the cluster autoscaler (https://github.com/kubernetes/autoscaler/pull/2101#issuecomment-499698555), this project doesn't allow for external dependencies other than those already in use within the project.

Reply:
I missed that. I guess you are stuck then. Seems kind of sad, but so be it. Thanks for clarifying.

Reply:
Might be good to bring that requirement back up at the next sig-autoscaling meeting, it seems like it introduces arbitrary complications, especially as support for cloud providers in k/k is in the process of being removed in favor of external cloud providers.

Reply:
I won't argue with that @detiber :-)

Is autoscaler going into a per-cloud-provider pluggable format, like the core cloud providers are doing? Or could we even have autoscaler somehow talk to the ccm itself?

Reply:
There are discussions around adopting a plugin-type model for cluster-autoscaler, such as https://github.com/kubernetes/autoscaler/pull/3140, but I don't know that there is consensus in that area currently.

I don't know that there could (or should) be direct integration with CCM, since CCM is generally scoped to a much smaller functional subset of cloud provider support than is used by cluster-autoscaler and adding support for scaling primitives needed for cluster-autoscaler seems a bit excessive for those components

Reply:
Got it. Overall, a nice plugin model would be great for this, if we could make it manageable. Keep all of this functionality in separate repos, and the core `kubernetes/autoscaler` repo is clean. Since cluster-api v1alpha3 switched to work this way, it has been great, much easier to maintain.